frontend/server/src/main/java/org/pytorch/serve/wlm/WorkerThread.java

```diff
@@ -184,28 +186,55 @@ public void run() {
                 req = aggregator.getRequest(workerId, state);
 
                 long wtStartTime = System.currentTimeMillis();
-                logger.info("Flushing req. to backend at: " + wtStartTime);
+                logger.info("Flushing req.cmd {} to backend at: {}", req.getCommand(), wtStartTime);
                 backendChannel.writeAndFlush(req).sync();
-
+                boolean isStreaming =
+                        req.getCommand() == WorkerCommands.STREAMPREDICT ? true : false;
+                ModelWorkerResponse reply = null;
+                long duration = 0;
                 long begin = System.currentTimeMillis();
-                ModelWorkerResponse reply = replies.poll(responseTimeout, TimeUnit.SECONDS);
-
-                long duration = System.currentTimeMillis() - begin;
-                logger.info("Backend response time: {}", duration);
-
-                if (reply != null) {
-                    aggregator.sendResponse(reply);
-                } else if (req.getCommand() != WorkerCommands.DESCRIBE) {
-                    int val = model.incrFailedInfReqs();
-                    logger.error("Number or consecutive unsuccessful inference {}", val);
-                    throw new WorkerInitializationException(
-                            "Backend worker did not respond in given time");
+
+                if (!isStreaming) {
+                    reply = replies.poll(responseTimeout, TimeUnit.SECONDS);
+
+                    duration = System.currentTimeMillis() - begin;
+                    logger.info("Backend response time: {}", duration);
+
+                    if (reply != null) {
+                        aggregator.sendResponse(reply);
+                    } else if (req.getCommand() != WorkerCommands.DESCRIBE) {
+                        int val = model.incrFailedInfReqs();
+                        logger.error("Number or consecutive unsuccessful inference {}", val);
+                        throw new WorkerInitializationException(
+                                "Backend worker did not respond in given time");
+                    }
+                } else {
+                    ModelInferenceRequest inferReq = (ModelInferenceRequest) req;
+                    boolean streamNext = true;
+                    while (streamNext) {
+                        reply = replies.poll(responseTimeout, TimeUnit.SECONDS);
```

Comment:
Looks like responseTimeout is same for streaming and non-streaming as well. clients might want different timeouts for streaming and non streaming api right?

Reply:
responseTimeout is planned to move to model level config.